gammapy/modeling/models/spatial.py

```diff
@@ -1105,6 +1117,12 @@ def to_dict(self, full_output=False):
         data["filename"] = self.filename
         data["normalize"] = self.normalize
         data["unit"] = str(self.map.unit)
+        if self.filename is None:
+            raise IOError("Missing filename")
+        elif os.path.isfile(self.filename) :
```

Comment:
Can we have a `overwrite` option rather? The original file on disc will be overwritten only if explicitly desired?

Reply:
added a `TemplateSpatialModel.write(overwrite=False)` method and added extra keyword `overwrite_templates` to `Models.write()`, `.to_dict()` and `.to_yaml()` to propagates this option.